pkg/flowaggregator/flowaggregator.go

```diff
@@ -374,6 +381,7 @@ func (fa *flowAggregator) sendFlowKeyRecord(key ipfixintermediate.FlowKey, recor
 	if err := fa.set.PrepareSet(ipfixentities.Data, templateID); err != nil {
 		return fmt.Errorf("error when preparing set: %v", err)
 	}
+	fa.fillK8sMetadata(key, record.Record)
```

Comment:
Doing this for every record is ok for now because we are deleting the record right after sending (L393). With PR #1949, we do not do that. Is there scope to improve this by identifying the fresh aggregated records?

Reply:
I'm thinking about adding a straightforward variable like `isMetadataFilled` to aggregated record. For each record to be exported, we just check whether `isMetadataFilled` is true or not. It can be set as true by default, and set to false when we have [correlation check](https://github.com/vmware/go-ipfix/blob/main/pkg/intermediate/aggregate.go#L763) in aggregation process. What do you think?